csrc/scheduler/registry.cpp

```diff
@@ -2119,11 +2128,13 @@ class PersistentKernelScheduler : public SchedulerEntry {
     return true;
   }
 
-  static std::pair<int64_t, int64_t> getPersistentBufferSize(
+  static std::tuple<int64_t, int64_t, bool> checkPersistentBufferSize(
       Fusion* fusion,
       SchedulerRuntimeInfo& runtime_info,
       HeuristicSummary* data_cache,
-      const std::vector<TensorView*>& reduction_tvs) {
+      const std::vector<TensorView*>& reduction_tvs,
+      const scheduler_utils::ReductionTvProperties& properties,
+      const int64_t vectorize_factor) {
```

Comment:
Why does this function belong in this file?

Reply:
It is a private function only used by `PersistentKernelScheduler`, I'll move it to `normalization_scheduler_utils`.